### PR TITLE
Add core files.

### PR DESCRIPTION
**Pull request by mattcollier**
Putting in PR now to get feedback on basic API etc before adding docs.

This is how a package is marked-up: https://github.com/digitalbazaar/bedrock-package-manager/blob/initial/test/mock-package/package.json#L1-L2

This is what the `find` API returns: https://github.com/digitalbazaar/bedrock-package-manager/blob/b8c34123dd4c3d425f557f0b7bc26b7d302c3953/test/mocha/01-api.js#L16-L20

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+const exec = promisify(require('child_process').exec);
+const path = require('path');
+const {util: {BedrockError}} = bedrock;
+
+const bedrockPackages = {};
+
+bedrock.events.on('bedrock.init', async () => {
+  let stdout;
+  try {
+    ({stdout} = await exec('npm ls --parseable --silent'));
```
Comment (davidlehn) [inline lib/index.js]: I'm concerned about the sure-to-be-annoying startup cost of doing this.  First run:
```
.../random-bedrock-project$ time npm ls --parseable --silent
real    0m8.183s
```
Caching or whatever will make it faster later:
```
real    0m2.250s
```
I think my setup is a bit quirky, so those may be high, but it's still something that will slow things down.

Also there is a `--json` output option.

Comment (davidlehn) [reply lib/index.js]: They also had an API, `npm.commands.ls(...)`, but maybe that was deprecated?  Docs not at their old location.

Comment (mattcollier) [reply lib/index.js]: I spot checked a few top level modules, and I'm seeing max value of ~5 sec with the mean being closer to 3 sec.  

Here is sample --json output which unfortunately does not include arbitrary properties from package.json files: https://gist.githubusercontent.com/mattcollier/203191d7407dd27dd6526eb6caf2f39e/raw/e89316c8c8059a5b6a8c7526efb27e44ca7fd185/some.json

After I found the following post indicating that the NPM nodejs API is a use at your own risk sort of thing, I abandoned the search: https://stackoverflow.com/a/15957574

Comment (dlongley) [reply lib/index.js]: Yikes -- waiting seconds for startup could easily get annoying fast, particularly when developing. Sounds like we should target another method to get the information we need or have a good mitigation strategy.

**Comment by mattcollier** [comment]
We're comparing module registration here with what we've done in the ledger stack: https://github.com/digitalbazaar/bedrock-ledger-node/blob/master/lib/rootApi.js#L24-L36

The injector in the ledger stack improves upon the injector being used in BSD code in that one does have the ability to specify a `type` for the plugin.

At this point, I'm not sure that the `npm ls` gyrations (with startup performance penalty) and package meta data is superior to having a `register` API in this module.

I do think that having `bedrock-package-manager` as a central repository for all plugins in the application is an improvement over `bedrock-injector` where various bedrock modules host their own plugin repository.

**Comment by dlongley** [comment]
> At this point, I'm not sure that the npm ls gyrations (with startup performance penalty) and package meta data is superior to having a register API in this module.
> 
> I do think that having bedrock-package-manager as a central repository for all plugins in the application is an improvement over bedrock-injector where various bedrock modules host their own plugin repository.

+1 to all of the above.

**Comment by mattcollier** [comment]
We decided to abandon the package meta data method.

**Pull request closed without merging**